File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = max(nums)
        temp_sum = 0
        for i in range(0 , len(nums)):
            if((temp_sum + nums[i])  >=  0 ):
                temp_sum += nums[i]
                max_sum = max(max_sum  , temp_sum)
            else:
                temp_sum = 0
        return (max_sum)

File: 53-maximum-subarray/maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
			int MaxSum = nums.Max();
		int dummy = 0;
			
			for (int i = 0; i < nums.Length; i++)
			{
				dummy += nums[i];
				if (dummy >= 0)
				{
					MaxSum = Math.Max(MaxSum, dummy);
				}
				else
				{
					dummy = 0;
				}

                // MaxSum = Math.Max(MaxSum, dummy);

			}
			
			return(MaxSum);


        
    }
}